deluge/common.py

```diff
@@ -961,24 +969,82 @@ def is_ipv6(ip):
 
     """
 
+    import socket
+
     try:
-        import ipaddress
-    except ImportError:
+        socket.inet_pton(socket.AF_INET6, ip)
+        return True
+    except OSError:
+        log.warning('Unable to verify IPv6 Address with socket.')
+    return False
+
+
+def is_net_interface(interface):
+    """A test to see if 'interface' is a valid network interface.
+
+    Args:
+        interface (str): The Interface to test. eg. eth0 linux. GUID on Windows.
+
+    Returns:
+        bool: Whether Interface is valid or not.
+
+    Examples:
+        >>> is_net_interface("eth0")
+        True
+        >>> is_net_interface("{7A30AE62-23ZA-3744-Z844-A5B042524871}")
+        True
+
+    """
+    if windows_check() and not interface.isupper():
```

Comment:
This is not something that a common function should care about and should not modify either

Reply:
I figured it would make it easyer on people to avoid simple errors and depending on how they find the network guid it may or may not provide it in upper and lead to confusion I could see people thinking "im putting it in right, why is deluge broken?" hence why it also logs a message so people know what it should be. Your call though ill take it out.

Reply:
I understand but we can do that verification in the UI where we can easily provide feedback to the user.